src/Mpdf.php

```diff
@@ -1587,12 +1593,51 @@ private function initConstructorParams(array $config)
 	private function initFontConfig(array $config)
 	{
 		$configObject = new FontVariables();
-		$defaults = $configObject->getDefaults();
+		$defaults     = $configObject->getDefaults();
+
+		$languageToFontRegister = new LanguageToFontRegistry([
+			'Core' => $this->languageToFont
+		]);
+
+		if (isset($config['fontRegistry'])) {
+			$fontRegistry = $config['fontRegistry'];
+
+			if ( ! $fontRegistry instanceof FontRegistry) {
+				throw new MpdfException('The fontRegistry config option should be of the class /Mpdf/Fonts/FontRegistry');
+			}
+
+			/* Merge the fontRegistry package information into the correct mPDF config structure */
+			foreach ($fontRegistry->getAll() as $package) {
+				$this->AddFontDirectory($package->getFontDir());
+
+				$packageFontData = [
+					'fontdata' => $package->getFontData(),
+					'BMPonly' => $package->getBmpFonts(),
+					'backupSubsFont' => $package->getBackupSubsFont(),
+				] + $package->getFontFamilySubstitution();
+
+				foreach( $packageFontData as $id => $val ) {
+					if (isset($config[$id])) {
+						$config[$id] = array_merge($config[$id], $val );
+					} else {
+						$config[$id] = array_merge($val, $defaults[$id]);
+					}
+				}
```

Comment:
`array_merge()` should be avoided in loops (and we're in two loops)

would it be possible to gather data in the loop instead, and perform a single array_merge at the end ? 🤔

Reply:
Not sure if that'll be possible. Would love it if you could take a closer look and brainstorm some ideas.